scripts/docker/install_airflow_dependencies_from_branch_tip.sh

```diff
@@ -51,7 +51,8 @@ function install_airflow_dependencies_from_branch_tip() {
     pip install --root-user-action ignore \
       ${ADDITIONAL_PIP_INSTALL_FLAGS} \
       "https://github.com/${AIRFLOW_REPO}/archive/${AIRFLOW_BRANCH}.tar.gz#egg=apache-airflow[${AIRFLOW_EXTRAS}]" \
-      --constraint "${AIRFLOW_CONSTRAINTS_LOCATION}" || true
+      --constraint "${AIRFLOW_CONSTRAINTS_LOCATION}" \
+      || pip install 'pydantic>=2' || true
```

Comment:
This hack (and those in Dockerfile and Dockerfile.ci, those are auto-copied by pre-commit) will be removed before merge.